vmdb/app/models/mixins/authentication_mixin.rb

```diff
@@ -92,8 +118,18 @@ def update_authentication(data, options = {})
       end
 
       # Update or create
-      cred = self.authentications.build(:name => "#{self.class.name} #{self.name}", :authtype => type.to_s, :type => "AuthUseridPassword") if cred.nil?
-      cred.userid, cred.password = value[:userid], value[:password]
+      if cred.nil?
+        if self.kind_of?(EmsOpenstackInfra) && value[:auth_key]
```

Comment:
@Ladas it would be to have a default authentication builder method that `EmsOpenstackInfra` could override.  It doesn't have to be in this PR but I think it would simplify this PR.